cmd/puppeth/wizard.go

```diff
@@ -104,40 +105,60 @@ func (w *wizard) readString() string {
 	}
 }
 
-// readYesNo reads a yes or no from stdin, returning boolean true for yes
-func (w *wizard) readYesNo(def bool) bool {
+// readDefaultString reads a single line from stdin, trimming if from spaces. If
+// an empty line is entered, the default value is returned.
+func (w *wizard) readDefaultString(def string) string {
+	fmt.Printf("> ")
+	text, err := w.in.ReadString('\n')
+	if err != nil {
+		log.Crit("Failed to read user input", "err", err)
+	}
+	if text = strings.TrimSpace(text); text != "" {
+		return text
+	}
+	return def
+}
+
+// readDefaultYesNo reads a single line from stdin, trimming if from spaces and
+// interpreting it as a 'yes' or a 'no'. If an empty line is entered, the default
+// value is returned.
+func (w *wizard) readDefaultYesNo(def bool) bool {
 	for {
 		fmt.Printf("> ")
 		text, err := w.in.ReadString('\n')
 		if err != nil {
 			log.Crit("Failed to read user input", "err", err)
 		}
 		text = strings.ToLower(strings.TrimSpace(text))
+		if text = text; text == "" {
```

Comment:
What does this construct do? test = test ? Why not just `if len(text) == 0`  ?

Reply:
Erm, that's a copy pate errors, should be `if text == "" {`

Reply:
Because all other code uses this construct and not the len version.